social_media_base/models/social_post.py

```diff
@@ -0,0 +1,273 @@
+                    f"social_media_{account.media_id.media_type}.social_media_post_preview",
+                    values | self._render_values_preview(),
+                )
+            except ValueError:
+                render_template += """\n\n""" + IrQweb._render(
+                    "social_media_base.social_media_post_preview",
+                    values | self._render_values_preview(),
+                )
+        return render_template if render_template else _("No preview available")
+
+    @api.depends("account_ids", "message", "image_ids", "video_ids")
+    def _compute_post_preview(self):
+        """
+        This method is responsible for obtaining the templates
+        to preview the posts when they are created.
+
+        Template ID format:
+         * social_media_{media_type}.social_media_post_preview
+        Example:
+         * social_media_linkedin.social_media_post_preview
+
+        As many templates as there are media according to the accounts selected in
+        the post will be rendered.
+
+        If the template does not exist, a default one is rendered.
+        """
+        for post in self:
+            post.post_preview = post._render_template_preview()
+
+    def _default_account_ids(self):
+        """
+        Returns a list of default account IDs to use when creating a post.
+        By default, this method returns an empty list.
+
+        :rtype: list[int]
+        """
+        return []
+
+    @api.model
+    def default_get(self, fields):
+        res = super().default_get(fields)
+        account_ids = self._default_account_ids()
+        if account_ids:
+            res["account_ids"] = [(6, 0, account_ids)]
+        return res
+
+    def action_create_post_account(self):
+        """
+        Publishes the post on all the
+        associated social media accounts.
+        """
+        self._action_create_post_account()
+
+    def _prepare_post_account_values(self):
+        """
+        Prepare the posting values for each account.
+        """
+        posts_account = []
+        for account in self.account_ids:
+            if account.id not in self.post_account_ids.mapped("account_id").ids:
+                posts_account.append(
+                    Command.create(
+                        {
+                            "post_id": self.id,
+                            "account_id": account.id,
+                            "state": "ready",
+                            "message": self.message,
+                        }
+                    )
+                )
+        return posts_account
+
+    def _action_create_post_account(self):
+        """
+        The posts are sent to social media.
+        """
+        for post in self:
+            post.write(
+                {
+                    "state": "publishing",
+                    "post_account_ids": post._prepare_post_account_values(),
+                }
+            )
+            post.post_account_ids[0]._action_post()
```

Comment:
This only executes `action_post` for the first `social.post.account`. I think it should run for all `social.post.account` records.
```suggestion
            for post_account in post.post_account_ids:
                post_account._action_post()
```